### ISSUE TITLE
Sprawdzanie warunków zwycięstwa magicznym kwadratem

### ISSUE THREAD

**Issue by event15** [CodersCommunity/zabawa-w-refaktoryzacje-challange-forumowy#2]
| 4 | 9 | 2 |
| --- | --- | --- |
| 3 | 5 | 7 |
| 8 | 1 | 6 |

To tak zwany magiczny kwadrat. Warunkiem zwycięstwa jest suma równa 15 w linii. 

Przykłady:
1. Poziomo: 4+9+2 = 3+5+7 = 8+1+6 = 15
2. Pionowo: 4+3+8 = 9+5+1 = 2+7+6 = 15
3. Diagonal: 4+5+6 = 8+5+2 = 15

W ten sposób sprawdzono wszystkie warunki wygranej a dla każdego gracza wystarczy sumować te liczby.

Tutaj dołączam kod, który może wygenerować taki magiczny kwadrat automatycznie (oczywiście do refaktoringu): http://code.geeksforgeeks.org/QeZ3yL

**Comment by efiku** [comment]
Rozumiem, że to jest pomysł który ma zastąpić te kosmiczne ify w kodzie do weryfikacji czy są 3 kółka / krzyżyki w poziomie , pionie , na skos ?

**Comment by event15** [comment]
Oczywiście, że tak. Dla planszy 3x3 kod sprowadza się do sprawdzenia jednego warunku. 
W przypadku, gdy plansza będzie dynamiczna i będzie mogła mieć dowolne rozmiary to jednak będzie konieczna implementacja metody generującej taki magiczny kwadrat. 

Jednak przykład z linku to full-stack metoda. A w naszym programie wystarczyłoby wyliczyć tylko sumę dla każdej linii. Myślę, że sam fakt odchudzi znacznie tą metodę.

**Comment by efiku** [comment]
Pozostanie tylko zrobienie tego tak, aby podmiana z prostego algo  na np ten, nie wprowadzała konieczności przebudowy całego sytemu ( ͡° ͜ʖ ͡°)

**Comment by event15** [comment]
I tak cały jest do przebudowy przecież. W tym momencie każde rozwiązanie jest mega dobre.
